src/negotiation/order.ts

```diff
@@ -16,8 +16,125 @@ export interface OrderAsset {
   nominalAmount: string;
 }
 
+export interface TakerCriteria {
+  buy: TakerCriteriaAsset;
+  sell: TakerCriteriaAsset;
+  minRate?: number;
+}
+
+export interface TakerCriteriaAsset {
+  ledger: string;
+  asset: string;
+  minNominalAmount?: string;
+  maxNominalAmount?: string;
+}
+
+/**
+ * @description An order handler for the taker. It has helper functions to facilitate the handler of an
+ * order by a taker. This should only be created via `TakerNegotiatior.getOrder()` and should not be constructed from
+ * scratch.
+ * @param orderParams - The raw parameters of the order.
+ * @param criteria - The criteria used to get this order.
+ * @param takeOrder - Function passed from the `TakerNegotiator` to coordinate calls to `cnd` and the maker to effectively
+ * take the order and start the atomic swap execution.
+ */
+export class Order {
+  /**
+   * @description **Note: This should not be used, `Order` should be created by using `TakerNegotiatior.getOrder()`
```

Comment:
Can one make the constructor private then and use `TakerNegotiatior.getOrder()` like a factory?

Reply:
If the constructor is private then **I**, a developer working on comit-sdk, cannot create `Order`. What you want is the TS equivalent Rust's `pub(crate)` which does not exist in TS as far as I know.